### PR TITLE
MDS-2287: Export Notice of Work

### PR DESCRIPTION
**Pull request by mhuseinov**
image
image
image
mds

### CODE REVIEW

**Review by nayr974 — COMMENTED** [review]
Looks good to me. Happy to approve once a few eyes have been on it!

File: migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+INSERT INTO document_template
+    (document_template_code, template_file_path, active_ind, create_user, update_user, form_spec_json)
+VALUES
+    ('NTR', 'templates/now/NOW Technical Review.docx', true, 'system-mds', 'system-mds', '[]');
+
```
Comment (mmumby) [inline migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql:5]: Is there any issues with the file path due to spaces in the name? Should there be a - or _?

File: services/core-api/app/api/now_applications/models/now_application_document_type.py [hunk 2]
```diff
@@ -1,11 +1,6 @@
-import uuid, datetime
-
-from sqlalchemy.ext.hybrid import hybrid_property
-from sqlalchemy.orm import validates
-from sqlalchemy.dialects.postgresql import UUID
 from sqlalchemy.schema import FetchedValue
```
Comment (mmumby) [inline services/core-api/app/api/now_applications/models/now_application_document_type.py:1]: all unused imports?

Comment (exzizt) [reply services/core-api/app/api/now_applications/models/now_application_document_type.py:1]: Yup!

File: migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql (hunk 1 above)
Comment (exzizt) [reply migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql:5]: Nope :)

File: services/core-api/app/date_time_helper.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from dateutil.relativedelta import relativedelta
+
+
+def get_duration_text(start_date, end_date):
```
Comment (mmumby) [inline services/core-api/app/date_time_helper.py:4]: Did you move this logic to the BE or duplicate the logic on the BE? Is there a chance they will ever return different values?

Comment (exzizt) [reply services/core-api/app/date_time_helper.py:4]: Mekhti wrote tests to ensure that the function works correctly. It's the same logic as the FE.

File: services/core-web/src/actionCreators/documentActionCreator.js [hunk 4]
```diff
@@ -69,4 +69,38 @@ export const generateNoticeOfWorkApplicationDocument = (
     .finally(() => dispatch(hideLoading("modal")));
 };
 
+export const exportNoticeOfWorkApplicationDocument = (
```
Comment (mmumby) [inline services/core-web/src/actionCreators/documentActionCreator.js:72]: Can you create a test for this?

**Review by mmumby — COMMENTED** [review]
Looks really good! Few small questions/comments

File: migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+INSERT INTO now_application_document_type
+    (now_application_document_type_code, description, active_ind, create_user, update_user, document_template_code)
+VALUES
+    ('NTR', 'Technical Review', true, 'system-mds', 'system-mds', 'NTR');
```
Comment (JustinMacaulay) [inline migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql:9]: Was this added to the seed data as well?

Comment (mhuseinov) [reply migrations/sql/V2020.09.22.15.30__added_now_technical_review_template.sql:9]: It is added now, thanks!

File: services/core-web/src/actionCreators/documentActionCreator.js (hunk 4 above)
Comment (mhuseinov) [reply services/core-web/src/actionCreators/documentActionCreator.js:72]: done

**Review by mmumby — DISMISSED** [review]
Good work! 🚢  it

**Review by mmumby — APPROVED** [review]

**Pull request merged**